liblouis/compileTranslationTable.c

```diff
@@ -4018,10 +4018,11 @@ compileRule(FileInfo *nested, CharacterClass **characterClasses,
 				ok = 0;
 		}
 		if (getToken(nested, &token, "character class name", &lastToken)) {
-			if ((class = findCharacterClass(&token, *characterClasses))) {
-				compileError(nested, "character class already defined.");
-			} else if ((class = addCharacterClass(nested, &token.chars[0], token.length,
-								characterClasses, characterClassAttribute))) {
+			class = findCharacterClass(&token, *characterClasses);
+			if (!class)
+				class = addCharacterClass(nested, &token.chars[0], token.length,
+						characterClasses, characterClassAttribute);
+			if (class) {
```

Comment:
Maybe it's me but this piece of code took me a while to understand. The code before seemed pretty straight forward. Maybe add a comment to the first if
``` C
// no class with that name: create one
```
then to the second if
``` C
// there is a class with that name or a new class was successfully created
```

Reply:
I thought the code after is just as straight forward as before. If you hadn't looked at the change, maybe it would have taken you a while to understand the old code too?

But anyway, some more comments never hurt, I'll add them. Thanks.